lib/preview_web/controllers/package_controller.ex

```diff
@@ -0,0 +1,27 @@
+defmodule PreviewWeb.PackageController do
+  use PreviewWeb, :controller
+
+  def raw(conn, %{"package" => package, "filename" => filename} = params) when filename != [] do
+    version = params["version"] || Preview.Bucket.get_latest_version(package)
+    filename = Path.join(filename)
+
+    if data = Preview.Bucket.get_file(package, version, filename) do
+      conn
+      |> maybe_put_content_type(Path.extname(filename))
+      |> put_resp_header("cache-control", "public, max-age=300")
```

Comment:
I'd like some advice on proper cache policy here. Given we have 24h on first publish and 1h on subsequent releases, perhaps we set max-age accordingly and otherwise 1yr or something? What'd be the most efficient way to figure out if it's first or subsequent release? I'd like to avoid making unnecessary API calls. I think hitting repo.hex.pm/packages/<name> would probably be the way to go?